diesel/src/query_builder/query_id.rs

```diff
@@ -7,15 +7,43 @@ use super::QueryFragment;
 /// All types which implement `QueryFragment` should also implement this trait
 /// (It is not an actual supertrait of `QueryFragment` for boxing purposes).
 ///
-/// This trait is generally implemented by invoking the [`impl_query_id!`]
-/// macro.
-///
 /// See the documentation of [the `QueryId` type] and [`HAS_STATIC_QUERY_ID`]
 /// for more details.
 ///
-/// [`impl_query_id!`]: ../macro.impl_query_id.html
 /// [the `QueryId` type]: #associatedtype.QueryId
 /// [`HAS_STATIC_QUERY_ID`]: #associatedconstant.HAS_STATIC_QUERY_ID
+///
+/// ### Deriving
+///
+/// This trait can be automatically derived by `diesel_derives`.
```

Comment:
We should not mention `diesel_derives` here, because this derive is provided by diesel itself for all users outside of diesel.

Reply:
Oh right hence why we can use it in Diesel >_<